### PR TITLE
Add new ItemEditableInfo

### PR DESCRIPTION
**Pull request by grosbouddha**
## Description

Create new ItemEditableinfo widget

image

## What has been done

Extending the base item and adding a readonly mode for the ItemEditableinfo non-editable case.

## How was it was tested

Locally in storybook

### CODE REVIEW

File: src/itemEditableInfo/index.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import React from 'react'
+
+import { TextDisplayType } from 'text'
+import Item from '_utils/item'
+import { A11yProps, pickA11yProps } from '_utils/interfaces'
+import { color } from '_utils/branding'
+
+export interface ItemEditableInfoProps extends A11yProps {
```
Comment (gabrielagabriel) [inline src/itemEditableInfo/index.tsx:8]: Are we keeping interfaces instead of types?

File: src/itemEditableInfo/index.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+const ItemEditableInfo = (props: ItemEditableInfoProps) => {
+  const { className, label, value, href = null, disabled = false } = props
+  const a11yAttrs = pickA11yProps<ItemEditableInfoProps>(props)
+
+  const extraProps = {
+    isClickable: true,
+    href,
+    // For the ItemEditableInfo, the value (entered previously by the user)
+    // is the most important info and is visually bigger than the
+    // label for it.
```
Comment (gabrielagabriel) [inline src/itemEditableInfo/index.tsx:31]: Not possible to replace for Typography?

File: src/itemEditableInfo/index.unit.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import React from 'react'
+import { mount } from 'enzyme'
+
+import ItemEditableIInfo from 'itemEditableInfo'
```
Comment (Tip-Sy) [inline src/itemEditableInfo/index.unit.tsx:4]: Typo: `ItemEditableInfo`
(fix below as well)

File: src/itemEditableInfo/index.unit.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+it('Should render a basic ItemEditableInfo', () => {
+  const props = {
+    label: 'My label',
+    value: 'My value',
+    href: '/myhref',
+  }
+  const wrapper = mount(<ItemEditableIInfo {...props} />)
+  const anchor = wrapper.find('a')
```
Comment (gabrielagabriel) [inline src/itemEditableInfo/index.unit.tsx:13]: Avoid targeting tags

File: src/itemEditableInfo/index.unit.tsx (hunk 3 above)
Comment (Tip-Sy) [reply src/itemEditableInfo/index.unit.tsx:4]: - [ ] fix typo

File: src/itemEditableInfo/index.tsx (hunk 1 above)
Comment (grosbouddha) [reply src/itemEditableInfo/index.tsx:8]: I need the old a11yProps pick mechanism :/

File: src/itemEditableInfo/index.tsx (hunk 2 above)
Comment (grosbouddha) [reply src/itemEditableInfo/index.tsx:31]: I would love to but I will need to recreate a new Item implmentation.
We need a separate project and PRs for that.

File: src/itemEditableInfo/index.unit.tsx (hunk 4 above)
Comment (grosbouddha) [reply src/itemEditableInfo/index.unit.tsx:13]: humm I do not agree, I need to verify that we have an anchor with a href and a good value.

File: src/itemEditableInfo/index.unit.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  expect(anchor.length).toBe(1)
```
Comment (wakooka) [inline src/itemEditableInfo/index.unit.tsx:14]: nit: toHaveLength(1)

File: src/itemEditableInfo/index.unit.tsx (hunk 3 above)
Comment (grosbouddha) [reply src/itemEditableInfo/index.unit.tsx:4]: Done

File: src/itemEditableInfo/index.unit.tsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+})
+
+it('Should render a non-editable ItemEditableInfo', () => {
+  const props = {
+    label: 'My label',
+    value: 'My value',
+    disabled: true,
+  }
+  const wrapper = mount(<ItemEditableIInfo {...props} />)
+  expect(wrapper.find('a').length).toBe(0)
```
Comment (wakooka) [inline src/itemEditableInfo/index.unit.tsx:26]: nit: `expect(wrapper.find('a).exists()).toBe(false)`

File: src/itemEditableInfo/index.tsx [hunk 7]
```diff
@@ -0,0 +1,53 @@
+import React from 'react'
+
+import { TextDisplayType } from 'text'
+import Item from '_utils/item'
+import { A11yProps, pickA11yProps } from '_utils/interfaces'
+import { color } from '_utils/branding'
+
+export interface ItemEditableInfoProps extends A11yProps {
+  readonly className?: Classcat.Class
```
Comment (wakooka) [inline src/itemEditableInfo/index.tsx:9]: After speaking with @simonrelet we have no reason to have this type, `className` should always be a string
`readonly className?: string`

File: src/itemEditableInfo/index.unit.tsx (hunk 4 above)
Comment (wakooka) [reply src/itemEditableInfo/index.unit.tsx:13]: I agree with @grosbouddha on this one but I think there's a need for a conversation on how we write test.

File: src/itemEditableInfo/index.unit.tsx (hunk 5 above)
Comment (grosbouddha) [reply src/itemEditableInfo/index.unit.tsx:14]: Done

File: src/itemEditableInfo/index.unit.tsx (hunk 6 above)
Comment (grosbouddha) [reply src/itemEditableInfo/index.unit.tsx:26]: Done

File: src/itemEditableInfo/index.tsx (hunk 1 above)
Comment (simonrelet) [reply src/itemEditableInfo/index.tsx:8]: This should work:

```ts
export type ItemEditableInfoProps = A11yProps & {
  // ...
}
```

File: src/itemEditableInfo/index.unit.tsx (hunk 4 above)
Comment (simonrelet) [reply src/itemEditableInfo/index.unit.tsx:13]: I added an entry for this in the next chapter weekly minutes. I only wrote my name but everyone is free to speak.

File: src/itemEditableInfo/index.tsx (hunk 1 above)
Comment (grosbouddha) [reply src/itemEditableInfo/index.tsx]: I updated it to the above, it works ! Thanks folks !

**Comment by grosbouddha** [comment]
FYI, I renamed the  **disabled** props to **readonly** which is closer to the widget spec.

**Pull request merged**